services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -287,6 +292,9 @@ async def remove_users_manually_marked_as_guests(
     guest_users: List[Tuple[int, str]] = await get_guest_user_ids_and_names(app)
     logger.info("GUEST user candidates to clean %s", guest_users)
 
+    # shuffle order of guest users to allow for random errors to not stop
+    # the removal of "not erroring" services
+    random.shuffle(guest_users)
```

Comment:
sorry, i do not see how changing the order in which users are deleted would prevent this error. At some point you will still find the faulty user, right?

Reply:
actually why not catching the error in line 446? and skipping that project while writting a warning in the log?

Reply:
> sorry, i do not see how changing the order in which users are deleted would prevent this error. At some point you will still find the faulty user, right?

You are correct. This dose not solve the issue, and it will still happen (as it should in my opinion, if something is wrong elsewhere)

Reply:
> actually why not catching the error in line 446? and skipping that project while writting a warning in the log?

It can be done, but it will only work on this case. The issue will still be occurring if some other call is broken.

Reply:
The idea of this changes is to not let the GC stuck over the same thing, but to still report the same issue and in the meanwhile keep cleaning.

Reply:
> > sorry, i do not see how changing the order in which users are deleted would prevent this error. At some point you will still find the faulty user, right?
> 
> You are correct. This dose not solve the issue, and it will still happen (as it should in my opinion, if something is wrong elsewhere)

Hmm but still I mean it's quite general no? if something bad happen with one study, you skip it on go on with the next one. in the end if anything fails with the APIs the GC calls it should skip that very project and log the error (and ideally with the project uuid)

Reply:
Let me have a go at refactoring so this can work properly.